### ISSUE TITLE
Add a Exif/Metadata editor.

### ISSUE THREAD

**Issue by ShapeShifter499** [SimpleMobileTools/Simple-Gallery#282]
I like this Gallery app, works pretty good. I would like to know though if there could be an Exif/Metadata editor. It would be nice to have to be able edit out sensitive location and camera data before posting some photos online.

**Comment by Michele64** [comment]
I would use this to add comments and to edit the location / gps info

**Comment by gappleto97** [comment]
I would like this

**Comment by jmichael2497** [comment]
i'm all for keeping Simple Gallery "simple".

for anyone looking to just remove EXIF data before sharing, try "Scrambled EXIF".

for anyone looking to do more extended EXIF editing, rating, and location mapping, try "A Photo Map".

both are found by searching the open source listings at https://search.f-droid.org/?q=exif

**Comment by kr428** [comment]
> i'm all for keeping Simple Gallery "simple".
> 

Agree with most of your suggestions - except for the "star" / favorite ratings. AFAICT SimpleGallery *has* the option to mark pictures as favorites so it only would make sense to allow for writing these information to the images files as well. :)

**Comment by Ivancaminal72** [comment]
As @kr428 said, there should be a metadata tag that saves the "star"/favorite information to the image file, in order to enable the recovery of that information from a computer (after a backup from the smartphone).

**Comment by jakommo** [comment]
@kr428 @Ivancaminal72 I've created a dedicated issue for storing the favourite/star rating in file https://github.com/SimpleMobileTools/Simple-Gallery/issues/1796 because I think it's only one part of what's requested here, but it might be much more simple to implement than an entire Exif/Metadata editor

**Comment by seniorm0ment** [comment]
Would like to see this as well, would be nice to just have a clear all metadata button too.

**Comment by szpak** [comment]
For people reaching our that issue, in the meantime, I started using [exif-eraser](https://github.com/Tommy-Geenexus/exif-eraser/). It removes exif metadata losslesly, it is FOSS, it doesn't require superfluous permissions and have a "share by default" option for people willing to just remove metadata before sending photos with some messanger.

**Comment by seniorm0ment** [comment]
yeah there are many apps it's just nice to have it built right in

**Comment by guaasamo** [comment]
there are many EXIF editor apps and there are many gallery apps, but there is no good app that does both

I use an app that can set EXIF descriptions for multiple selected photos 

please consider adding this feature to the pro version of gallery

thanks

**Comment by Donkey-Doug** [comment]
> there are many EXIF editor apps and there are many gallery apps, but there is no good app that does both
> 

Have a look at Aves gallery:
https://github.com/deckerst/aves

**Comment by tibbi** [comment]
starting with 6.23.0 it will be possible to remove EXIF values from the properties dialog

**Comment by szpak** [comment]
> starting with 6.23.0 it will be possible to remove EXIF values from the properties dialog

Great. Just please to remember to keep the orientation tag from Exif in the exported file ;).

**Comment by tibbi** [comment]
ye, orientation is the only one that wont be removed. There are likely many more tags that we could be removing, but that will improve over time.

**Comment by tibbi** [comment]
Actually there is a change, the EXIF remover wont be visible in the gallery on Android 11+ due to Googles strictened permissions, so this is joining the list of features that wont be available on those devices anymore. It will be there on lower OS versions or the file manager.

**Comment by tibbi** [comment]
if you find a gallery that can update EXIF in any way on Android 11+ and it was updated since november 2021, let me know.

**Comment by szpak** [comment]
> Actually there is a change, the EXIF remover wont be visible in the gallery on Android 11+ due to Googles strictened permissions

That's pity :-/.

> if you find a gallery that can update EXIF in any way on Android 11+ and it was updated since november 2021, let me know.

Does it have to be a gallery? exif-eraser seems to work on my Android 12 and it was updated in November 2021:
https://github.com/Tommy-Geenexus/exif-eraser/releases/tag/v5.0.0

**Comment by tibbi** [comment]
right, they are recreating the whole file, we might add it like that too. I wanted to avoid it though, it could cause additional issues and the UX is pretty bad.

**Comment by szpak** [comment]
Not always people would like to permanently remove the exif metadata from their photos. Personally, I do it before sending them via some app (e.g. social networks), so (under a hood) I make a copy of it (similarly to the photo edits). In fact, I would like to just remove exif and send that image to some other app and discard afterward. However, I remember that an author of exif-eraser told me, it is problematic (some temporary storage would need to be freed up periodically), so "exporting" them in a dedicated  "no-exif" folder (with an option to automatically share with some other app) is ok for me (I can purge the folder occasionally).

**Comment by tibbi** [comment]
this change I wrote above is a permanent change, some function like "Remove EXIF and Share" is a whole different thing. It could be a pretty useful feature together with "Resize and Share"

**Comment by jmichael2497** [comment]
> In fact, I would like to just remove exif and send that image to some other app and discard afterward. However, I remember that an author of exif-eraser told me, it is problematic (some temporary storage would need to be freed up periodically), so "exporting" them in a dedicated "no-exif" folder (with an option to automatically share with some other app) is ok for me (I can purge the folder occasionally).

you might be interested in https://f-droid.org/packages/de.kaffeemitkoffein.imagepipe which, yes makes a folder that does make its own folder, but easy enough to select all and delete with Gallery as needed.

meanwhile, i'm staying with 6.22.3 since i don't want to lose hidden folder view and other stuff that Google is making frustratingly difficult for end users and developers to do.
